centchain/extrinsic_status_task.go

```diff
@@ -169,9 +175,43 @@ func (est *ExtrinsicStatusTask) processRunTask() (resp interface{}, err error) {
 		return nil, gocelery.ErrTaskRetryable
 	}
 
-	// TODO(miguel) Add extrinsic success/failure status check
+	meta, err := est.getMetadataLatest()
+	if err != nil {
+		return nil, err
+	}
+
+	key, err := types.CreateStorageKey(meta, "System", "Events", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var er types.EventRecordsRaw
+	err = est.getStorage(key, &er, nhBlock)
+	if err != nil {
+		return nil, err
+	}
+
+	e := types.EventRecords{}
+	err = er.DecodeEventRecords(meta, &e)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check success events
+	for _, es := range e.System_ExtrinsicSuccess {
```

Comment:
based on this logic: https://github.com/polkadot-js/api/blob/master/packages/api/src/util/filterEvents.ts